### DISCUSSION TITLE
Obtaining ODF from XRD pole figures data

### DISCUSSION THREAD

**Discussion by ripudaman1047** [mtex-toolbox/mtex#1745] [Ask Anything] [answered]
[XRD_data.zip](https://github.com/mtex-toolbox/mtex/files/10814023/XRD_data.zip)

Hi, 
I am beginner undergraduate student using MTEX for a research project. I have got XRD pole figure data of an aluminium alloy, in ".txt" and ".asc" files (attached as a zipped file). I want to obtain ODF from this data, and then calculate orientations and obtain discrete orientations using ".Euler" method. 

- Which files type do I need to input in the import wizard and in what format. 

 I plan to use ".cif" file for aluminium for crystal symmetry and "mmm" as specimen symmetry. 

- Also, how do I know the symmetries I use are correct, as I haven't done the XRD myself, and was only handed the results.

**Comment by kilir** [comment, marked as answer]
Hi,
the import wizard guides me nicely through the prcedure after loading the three asc file, inlcuding the cif-file, specifying the specimen symmetry etc. Did you try it?

<img width="595" alt="output" src="https://user-images.githubusercontent.com/11697557/221257559-36ec1457-f169-4f50-95be-16726b10a9df.png">

In oder to compute an odf, please consult the documentation.

Cheers,
Rüdiger

**Comment by AmanKorea** [comment]
Dear Rudiger, 

Similar type of issue I am also facing, when while running the Mtex script, it shows "could not detect the file format".

![Capture1](https://github.com/mtex-toolbox/mtex/assets/141556865/40d6cd3d-6313-42d6-b42d-b158658f710b)

This is the file of pole figure in .asc format. 

[Asc_Script.zip](https://github.com/mtex-toolbox/mtex/files/12269217/Asc_Script.zip)

Can you guide me. 

Regards
Dr. Aman

**Comment by ralfHielscher** [comment]
For some reason your script contain the lines
```matlab
pftr = loadOrientation(fname,h,CS,SStr);
pf = loadOrientation(fname,h,CS,SS);
```
As you import pole figures it should be
```matlab
pftr = PoleFigure.load(fname,h,CS,SStr);
pf = PoleFigure.load(fname,h,CS,SS);
```
Ralf.

**Reply by AmanKorea** [reply]
Thank You. Problem is resolved now.

Aman
